easypy/caching.py

```diff
@@ -171,29 +172,73 @@ def clear():
 
     return deco
 
+if sys.version_info < (3, 5):
+    def _apply_defaults(bound_arguments):
+        """
+        Set default values for missing arguments. (from Python3.5)
+        """
+        from collections import OrderedDict
+        from inspect import _empty, _VAR_POSITIONAL, _VAR_KEYWORD
+
+        arguments = bound_arguments.arguments
+        new_arguments = []
+        for name, param in bound_arguments._signature.parameters.items():
+            try:
+                new_arguments.append((name, arguments[name]))
+            except KeyError:
+                if param.default is not _empty:
+                    val = param.default
+                elif param.kind is _VAR_POSITIONAL:
+                    val = ()
+                elif param.kind is _VAR_KEYWORD:
+                    val = {}
+                else:
+                    # This BoundArguments was likely produced by
+                    # Signature.bind_partial().
+                    continue
+                new_arguments.append((name, val))
+        bound_arguments.arguments = OrderedDict(new_arguments)
+else:
+    def _apply_defaults(bound_arguments):
+        return bound_arguments.apply_defaults()
```

Comment:
`bound_arguments.apply_defaults()` returns `None`, but it's still weird that this version "returns" something while the backward-compatibility version doesn't.

Reply:
I'm like a decorator, don't care about implementation detail, I want to be future-compatible.

Reply:
Future-compatibility-wise, if in the future apply_default will return something, someone might use the value returned from _apply_defaults believing it's backward compatible